pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -59,10 +66,43 @@ def _get_content_from_repo(repo_version_pk: str):
     content_queue.put(None)  # signal that there is no more content_units
 
 
+def _parse_npm_pkg_dependencies(package_lock_content):
+    """
+    Parse the package-lock.json file to extract the packages[name][dependencies] and
+    add them to content_queue Queue
+
+    notes:
+    - we are striping the "~" and "^" from versions because osv.dev has no support to version range
+    - the old/legacy packages[dependencies] field is not supported
+    """
+    for pkg in package_lock_content.get("packages", None):
+        if not package_lock_content["packages"][pkg].get("dependencies", None):
+            continue
+        for package_name, package_version in package_lock_content["packages"][pkg][
+            "dependencies"
+        ].items():
+            # we will not handle version range yet, for now, we will consider
+            # only the specific version
+            package_version = package_version.strip("^~")
+            content_queue.put(
+                {"name": package_name, "version": package_version, "ecosystem": PKG_ECOSYSTEM.npm}
+            )
+    content_queue.put(None)  # signal that there is no more content_units
+
+
 def _build_osv_data(content):
     """
     Helper function to build the osv.dev request data based on content object
     """
+
+    if isinstance(content, dict):
```

Comment:
Similar here I'd not overload this with another if case to handle. Start by making it totally separate and then refactor the common parts into yet a new function that both functions call into (the reusable part).